File: Agency2026MCP/Models/Contract.cs
namespace Agency2026MCP.Models
{
    public class Contract
    {
        // Identity
        public string ContractNumber { get; set; } = string.Empty;
        public string BaseContractNumber { get; set; } = string.Empty;
        public string? AmendmentCode { get; set; }
        public bool IsAmendment => !string.IsNullOrWhiteSpace(AmendmentCode);
        public bool isSoleSource { get; set; }
        // Justification
        public string? PermittedSituation { get; set; } // normalized to lowercase
      
        // Parties
        public string Department { get; set; } = string.Empty;
        public string DepartmentAddress { get; set; } = string.Empty;
        public string VendorName { get; set; } = string.Empty;
        public string VendorAddress { get; set; } = string.Empty;
       
        // Contract body
        public string Services { get; set; } = string.Empty;
        public decimal Value { get; set; } // This is the value of the contract if ammendment it is the value each ammenment added to the contract
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public string FiscalYear { get; set; } = string.Empty;
    }
}

File: FrontUI/procurementUI/src/types/Contract.ts
export interface Contract {
  // Identity
  contractNumber: string;
  baseContractNumber: string;
  amendmentCode?: string | null;
  isAmendment: boolean;
  isSoleSource: boolean;

  // Justification
  permittedSituation?: string | null;

  // Parties
  department: string;
  departmentAddress: string;
  vendorName: string;
  vendorAddress: string;

  // Contract body
  services: string;
  value: number;
  startDate?: string | null; // ISO string
  endDate?: string | null;
  fiscalYear: string;
}